python/cuml/internals/mem_type.py

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from enum import Enum, auto
+from cuml.internals.device_support import GPU_ENABLED
+from cuml.internals.safe_imports import (
+    cpu_only_import,
+    gpu_only_import
+)
+
+cudf = gpu_only_import('cudf')
+cp = gpu_only_import('cupy')
+cpx_sparse = gpu_only_import('cupyx.scipy.sparse')
+np = cpu_only_import('numpy')
+pandas = cpu_only_import('pandas')
+scipy_sparse = cpu_only_import('scipy.sparse')
+
+
+class MemoryTypeError(Exception):
+    '''An exception thrown to indicate inconsistent memory type selection'''
+
+
+class MemoryType(Enum):
+    device = auto(),
```

Comment:
Since enums are constants, they are typically capitalized.
```suggestion
    DEVICE = auto()
```

Reply:
I think I'm leaning toward lower-case since we want to keep the string representations lower-case, and I'd slightly prefer to have them match the enum value name. I can certainly be talked out of that though if you feel strongly.